File: sentimental-readability/readability.py
# TODO
def main():
    text = input("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters * 100 / words
    S = sentences * 100 / words

    index = 0.0588 * L - 0.296 * S - 15.8
    print(f"Grade: {round(index)}")

def count_letters(text):
    count = 0
    for letter in text:
        if letter.isalpha():
            count += 1
    return count

def count_words(text):
    count = 0
    for letter in text:
        if letter == " ":
            count += 1
    count += 1
    return count

def count_sentences(text):
    count = 0
    for letter in text:
        if letter in [".", "!", "?"]:
            count += 1
    return count


if __name__ == "__main__":
    main()

File: week2/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Getting text from user

    string text = get_string("Text: \n");

    // Counting letters in given text

    int letters = count_letters(text);

    // Counting words in given text

    int words = count_words(text);

    // Counting sentences in given text

    int sentences = count_sentences(text);

    float L = (float)letters / words * 100;
    float S = (float)sentences / words * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Printing the Grades
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        int grade = round(index);
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{

    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{

    int words = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{

    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '?' || text[i] == '!' || text[i] == '.')
        {
            sentences++;
        }
    }
    return sentences;
}